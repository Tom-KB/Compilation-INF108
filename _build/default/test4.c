
void f(int x){
    int y;
    y=3; 
   {y=2;} 
    print_int(y); 
     
} 

void main(int x){
	f(3); 
}
