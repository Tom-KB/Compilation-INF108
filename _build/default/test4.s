	.text
	j	main
	addi	$sp,$sp,-4
f:
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	move	$a0, $ra
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	li	$a0, 10
	li	$v0, 1
	syscall
	li	$v0, 11
	li	$a0, 10
	syscall
	li	$a0, 2
	sw	$a0,8($sp)
	lw	$ra,0($sp)
	addi	$sp,$sp,8
	jr	$ra
main:
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	move	$a0, $ra
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	li	$a0, 3
	jal	f
	lw	$ra,0($sp)
	addi	$sp,$sp,8
	jr	$ra
	li	$v0, 10
	syscall
	.data
