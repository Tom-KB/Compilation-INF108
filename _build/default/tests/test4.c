int main(int w)
{
    int x;
    int y;
    x = 0;
    while (x < 5) {
      y = 0;
      while (y < 5) {
        if (x+y == 3) break;
        print_int(x+y);
        y = y+1;
      }
      print_int(999);
      x=x+1;
    }


    return 0;
}