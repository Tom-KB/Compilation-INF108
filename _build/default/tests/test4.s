	.text
	j	main
main:
	move	$a0, $ra
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	addi	$sp,$sp,-4
	addi	$sp,$sp,-4
	li	$a0, 0
	sw	$a0,4($sp)
while1:
	lw	$a0,4($sp)
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	li	$a0, 5
	lw	$a1,0($sp)
	addi	$sp,$sp,4
	slt	$a0,$a1,$a0
	beq	$a0,$zero,done1
	li	$a0, 0
	sw	$a0,0($sp)
while2:
	lw	$a0,0($sp)
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	li	$a0, 5
	lw	$a1,0($sp)
	addi	$sp,$sp,4
	slt	$a0,$a1,$a0
	beq	$a0,$zero,done2
	lw	$a0,4($sp)
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	lw	$a0,4($sp)
	lw	$a1,0($sp)
	addi	$sp,$sp,4
	add	$a0,$a1,$a0
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	li	$a0, 3
	lw	$a1,0($sp)
	addi	$sp,$sp,4
	sub	$a0,$a1,$a0
	sltiu	$a0,$a0,1
	beq	$a0,$zero,suite1
	j	done2
suite1:
	lw	$a0,4($sp)
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	lw	$a0,4($sp)
	lw	$a1,0($sp)
	addi	$sp,$sp,4
	add	$a0,$a1,$a0
	li	$v0, 1
	syscall
	li	$v0, 11
	li	$a0, 10
	syscall
	lw	$a0,0($sp)
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	li	$a0, 1
	lw	$a1,0($sp)
	addi	$sp,$sp,4
	add	$a0,$a1,$a0
	sw	$a0,0($sp)
	j	while2
done2:
	li	$a0, 999
	li	$v0, 1
	syscall
	li	$v0, 11
	li	$a0, 10
	syscall
	lw	$a0,4($sp)
	addi	$sp,$sp,-4
	sw	$a0,0($sp)
	li	$a0, 1
	lw	$a1,0($sp)
	addi	$sp,$sp,4
	add	$a0,$a1,$a0
	sw	$a0,4($sp)
	j	while1
done1:
	li	$a0, 0
	addi	$sp,$sp,8
	lw	$ra,0($sp)
	addi	$sp,$sp,8
	jr	$ra
	addi	$sp,$sp,8
	lw	$ra,0($sp)
	addi	$sp,$sp,8
	jr	$ra
	li	$v0, 10
	syscall
	.data
