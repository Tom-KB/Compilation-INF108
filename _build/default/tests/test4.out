0
1
2
999
1
2
999
2
999
999
4
5
6
7
8
999
